src/main/java/fi/aalto/cs/apluscourses/model/Authentication.java

```diff
@@ -0,0 +1,15 @@
+package fi.aalto.cs.apluscourses.model;
+
+import fi.aalto.cs.apluscourses.utils.CoursesClient;
+import org.jetbrains.annotations.NotNull;
+
+public interface Authentication extends CoursesClient.HttpAuthentication {
+
+  void clear();
+
+  int maxTokenLength();
+
+  void setToken(@NotNull char[] newToken);
+
+  boolean isSet();
+}
```

Comment:
Is there a benefit in having this interface instead of `APlusAuthentication` directly extending `CoursesClient.HttpAuthentication`?

Reply:
Not really, I guess (except what is the usual benefit of having an interface). Actually, we need an interface that has `addToRequest()` and `isSet()` methods but which can be either token-authentication or password-authentication. So I'll change this to be that interface and let `APlusAuthentication` stay as a class for token authentication.

Reply:
Sounds good, then I guess `HttpAuthentication` can be removed if this will have `addToRequest`